runtime/binding-kafka/src/main/java/io/aklivity/zilla/runtime/binding/kafka/internal/stream/KafkaClientFetchFactory.java

```diff
@@ -3064,6 +3073,12 @@ private void onDecodeFetchResponse(
 
                 if (topicPartitions.get(partitionId) == leaderId)
                 {
+                    fetchRequestCount++;
+                    if (fetchRequestCount == 2 && !dataFrameSent)
+                    {
+                        client.triggerFlushIfRequired(traceId);
+                    }
+                    dataFrameSent = false;
```

Comment:
Let `client.triggerFlushIfRequired` implement the logic to determine what to do, and keep this part simple.